### PR TITLE
[ACS-4586] Implement Playwright Framework and add it to CI/CD (travis)

### PR DESCRIPTION
**Pull request by KikoUK**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [x] Other... Please describe:
See title

**What is the current behaviour?** (You can also link to an open issue here)
See title

**What is the new behaviour?**
See title

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: e2e/playwright/shared/base-config/global.setup.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  const browser = await chromium.launch();
+  const page = await browser.newPage();
+  const loginPage = new LoginPage(page);
+
+  try {
+    fs.mkdirSync(`./e2e/playwright/tests/folder-rules/tests/asd`, { recursive: true });
+    await page.goto('http://localhost:4200/login');
+    await console.log(baseUrl);
+    await console.log(acsAdminUser);
```
Comment (DenysVuika) [inline e2e/playwright/shared/base-config/global.setup.ts:41]: we should not dump this into the public logs I think

File: e2e/playwright/shared/base-config/global.setup.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    await loginPage.loginUser({ username: acsAdminUser, password: acsAdminUserPassword }, { withNavigation: false, waitForLoading: true });
+    await console.log(
+      '------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------'
```
Comment (DenysVuika) [inline e2e/playwright/shared/base-config/global.setup.ts:44]: I suggest removing this separator and just use space lines

File: e2e/playwright/shared/base-config/global.setup.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    );
+    await page.context().storageState({ path: `./e2e/playwright/tests/folder-rules/tests/asd/AdminUserState.json` });
+    await browser.close();
+  } catch (e) {
+    await console.log(
+      '+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++'
```
Comment (DenysVuika) [inline e2e/playwright/shared/base-config/global.setup.ts:50]: I suggest removing this separator and just use space lines

File: e2e/playwright/shared/base-config/global.setup.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  try {
+    fs.mkdirSync(`./e2e/playwright/tests/folder-rules/tests/asd`, { recursive: true });
+    await page.goto('http://localhost:4200/login');
+    await console.log(baseUrl);
+    await console.log(acsAdminUser);
+    await loginPage.loginUser({ username: acsAdminUser, password: acsAdminUserPassword }, { withNavigation: false, waitForLoading: true });
+    await console.log(
+      '------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------'
+    );
+    await page.context().storageState({ path: `./e2e/playwright/tests/folder-rules/tests/asd/AdminUserState.json` });
```
Comment (DenysVuika) [inline e2e/playwright/shared/base-config/global.setup.ts:46]: what's the point saving this between the runs?

File: e2e/playwright/tests/folder-rules/page-objects/components/actions-dropdown.component.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  private static rootElement = '.mat-select-panel';
+
+  public getOptionLocator = (optionName: string): Locator => this.getChild('.mat-option-text', { hasText: optionName });
+
+  constructor(page: Page) {
+    super(page, ActionsDropdownComponent.rootElement);
+  }
+
+  async selectAction(action: Partial<ActionType>): Promise<void> {
+    await this.page.locator(`aca-edit-rule-dialog [data-automation-id="rule-action-select"]`).click();
```
Comment (DenysVuika) [inline e2e/playwright/tests/folder-rules/page-objects/components/actions-dropdown.component.ts:28]: all automation ids are unique, so you can omit having parent selectors like `aca-edit-rule-dialog`

File: e2e/playwright/tests/folder-rules/page-objects/components/manage-rules-dialog.component.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+import { BaseComponent } from '@alfresco/playwright-shared';
+import { Page } from '@playwright/test';
+
+export class ManageRulesDialogComponent extends BaseComponent {
+  private static rootElement = 'aca-edit-rule-dialog';
+
+  public createRuleButton = this.getChild('[data-automation-id="edit-rule-dialog-submit"]');
+  private ruleNameInputLocator = this.getChild('[id="rule-details-name-input"]');
```
Comment (DenysVuika) [inline e2e/playwright/tests/folder-rules/page-objects/components/manage-rules-dialog.component.ts:16]: this id probably needs to be converted to the automation id 🤔 cc @TomikaArome

File: e2e/playwright/tests/folder-rules/playwright.config.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  projects: [
+    {
+      name: 'chromium',
+      use: {
+        ...devices['Desktop Chrome']
+      }
+    }
+  ]
+
+  /* Run your local dev server before starting the tests */
```
Comment (DenysVuika) [inline e2e/playwright/tests/folder-rules/playwright.config.ts:55]: not needed

File: package.json [hunk 8]
```diff
@@ -71,6 +71,10 @@
     "@angular/cli": "14.1.2",
     "@angular/compiler-cli": "14.1.2",
     "@angular/language-service": "14.1.2",
+    "@playwright/test": "^1.30.0",
+    "@reportportal/agent-js-playwright": "^5.0.8",
```
Comment (DenysVuika) [inline package.json:75]: this one not needed if you don't import or use it directly

File: package.json [hunk 9]
```diff
@@ -71,6 +71,10 @@
...
+    "allure-playwright": "^2.0.0-beta.15",
```
Comment (DenysVuika) [inline package.json:76]: not needed probably, as I don't see where it's imported in the code

File: package.json [hunk 10]
```diff
@@ -71,6 +71,10 @@
...
+    "eslint-plugin-playwright": "^0.9.0",
```
Comment (DenysVuika) [inline package.json:77]: this one is not used as well

File: scripts/ci/job_hooks/before_e2e.sh [hunk 11]
```diff
@@ -19,6 +19,9 @@ FROM=$1;
 TO=$2;
 PARAMS=$3;
 
+rm -rf node_modules/
```
Comment (DenysVuika) [inline scripts/ci/job_hooks/before_e2e.sh:22]: you can run `npm ci` that does exactly that, removing node modules and running install

File: scripts/ci/jobs/affected-project-with.sh [hunk 12]
```diff
@@ -17,18 +18,36 @@ target_options(){
...
 while [[ $1  == -* ]]; do
     case "$1" in
       -target)  project_target $2; shift 2;;
       -options)  target_options $2; shift 2;;
+      -test-runner)  test_runner $2; shift 2;;
       -*) echo "invalid option: $1" 1>&2; show_help; exit 0;;
     esac
 done
 
+#npm i -D @playwright/test
```
Comment (DenysVuika) [inline scripts/ci/jobs/affected-project-with.sh:34]: not needed

File: scripts/ci/jobs/affected-project-with.sh [hunk 13]
```diff
@@ -17,18 +18,36 @@ target_options(){
...
+npx playwright install chromium
 
-echo "Run alfresco-content-e2e protractor with options $OPTIONS"
-echo "./node_modules/.bin/protractor \"./protractor.conf.js\" $OPTIONS || exit 1"
+search_dir=./node_modules
```
Comment (DenysVuika) [inline scripts/ci/jobs/affected-project-with.sh:37]: it will be a giant list of node_modules contents, is that needed?

File: tsconfig.json [hunk 14]
```diff
@@ -12,12 +12,14 @@
...
     "noUnusedLocals": true,
+    "skipLibCheck": true,
     "noUnusedParameters": true,
     "noImplicitReturns": true,
     "target": "es2020",
     "resolveJsonModule": true,
     "allowSyntheticDefaultImports": true,
-    "typeRoots": ["node_modules/@types"],
+    "typeRoots": ["node_modules/@types",
+      "node_modules/commander/typings/index.d.ts"],
```
Comment (DenysVuika) [inline tsconfig.json:22]: probably not needed

**Comment by KikoUK** [comment]
Hey @DenysVuika. Thank you for your comments 👍

90% of the code (apart from the id selectors comments) mentioned in the comments above is all added to debug failing travis build where the logging state of the user (saving the cookies, a playwright feature) isn't populated in the storageState file which works fine when ran locally. That code will be removed once we have a green build ✅

**Pull request closed without merging**